superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FilterScope/FilterScope.tsx

```diff
@@ -56,11 +57,41 @@ const FilterScope: FC<FilterScopeProps> = ({
   scope,
   updateFormValues,
   chartId,
+  initiallyExcludedCharts,
 }) => {
-  const initialScope = scope || getDefaultScopeValue(chartId);
-  const initialScoping = isScopingAll(initialScope, chartId)
-    ? Scoping.all
-    : Scoping.specific;
+  const [initialScope] = useState(
+    scope || getDefaultScopeValue(chartId, initiallyExcludedCharts),
+  );
+  const [initialScoping] = useState(
+    isScopingAll(initialScope, chartId) ? Scoping.all : Scoping.specific,
+  );
```

Comment:
This isn't directly related to this PR, but it confused me while reading this code: having two props called `initialScope` and `initialScoping` right next to each other feels very ambiguous. If changing this to something slightly more specific doesn't lead down a huge rabbit hole, I'd opt for renaming `Scoping` to `ScopingType` and consequently `initialScoping` to `initialScopingType`, perhaps even `initialScope` to `initialChartScope`.

Reply:
Totally agree, I feel like I spend more time trying to wrap my head around difference between those 2 than actual developing. I'll change that in this PR